internal/proto/local/controller/api/resources/credentialstores/v1/credential_store.proto

```diff
@@ -0,0 +1,87 @@
+
+  // Optional user-set description for identification purposes.
+  google.protobuf.StringValue description = 50 [(custom_options.v1.generate_sdk_option) = true, (custom_options.v1.mask_mapping) = { this: "description" that: "Description" }];
+
+  // Output only. The time this resource was created.
+  google.protobuf.Timestamp created_time = 60 [json_name = "created_time"];
+
+  // Output only. The time this resource was last updated.
+  google.protobuf.Timestamp updated_time = 70 [json_name = "updated_time"];
+
+  // Version is used in mutation requests, after the initial creation, to ensure this resource has not changed.
+  // The mutation will fail if the version does not match the latest known good version.
+  uint32 version = 80;
+
+  // The Credential Store type.
+  string type = 90;
+
+  // The attributes that are applicable for the specific Credential Store type.
+  google.protobuf.Struct attributes = 100 [(custom_options.v1.generate_sdk_option) = true];
+
+  // Output only. The available actions on this resource for this user.
+  repeated string authorized_actions = 300 [json_name = "authorized_actions"];
+
+  // Output only. The authorized actions for the scope's collections.
+  map<string, google.protobuf.ListValue> authorized_collection_actions = 310 [json_name = "authorized_collection_actions"];
+}
+
+// The attributes of a vault typed Credential Store.
+message VaultCredentialStoreAttributes {
+  // The complete url address of vault.
+  google.protobuf.StringValue address = 10 [json_name = "address", (custom_options.v1.generate_sdk_option) = true, (custom_options.v1.mask_mapping) = { this: "attributes.address" that: "VaultAddress" }];
+
+  // The namespace of vault used by this store
+  google.protobuf.StringValue namespace = 20 [json_name = "namespace", (custom_options.v1.generate_sdk_option) = true, (custom_options.v1.mask_mapping) = { this: "attributes.namespace" that: "Namespace" }];
+
+  // The PEM encoded CA cert to verify the vault server's ssl certificate
+  google.protobuf.StringValue ca_cert = 30 [json_name = "ca_cert", (custom_options.v1.generate_sdk_option) = true, (custom_options.v1.mask_mapping) = { this: "attributes.ca_cert" that: "CaCert" }];
+
+  // The value to use as the SNI host when connecting to vault via TLS.
+  google.protobuf.StringValue tls_server_name = 40 [json_name = "tls_server_name", (custom_options.v1.generate_sdk_option) = true, (custom_options.v1.mask_mapping) = { this: "attributes.tls_server_name" that: "TlsServerName" }];
+
+  // When set to true verification of the TLS certificate is disabled.
+  google.protobuf.BoolValue tls_skip_verify = 50 [json_name = "tls_skip_verify", (custom_options.v1.generate_sdk_option) = true, (custom_options.v1.mask_mapping) = { this: "attributes.tls_skip_verify" that: "TlsSkipVerify" }];
+
+  // Input only field. The current vault token used by this credential store for creating new credentials.
```

Comment:
Down here you switch to "Input only field" and "Output only field" instead of "Output only" in the message above. Will grpc-gateway interpret that correctly? Not sure if it must literally be `Output only.` or if it just looks for the existence of those two words regardless of what follows them. @talanknight